### PR TITLE
Fix UnsetEnv for buildah bud

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by flouthoc** [comment]
LGTM. @rhatdan could you rebase. I clicked on `github UI` **rebase** and its adding another commit to the PR.

**Comment by flouthoc** [comment]
This is not a blocker.
One doubt which i have is that its only modifying `oci spec` but not the actual image. But I'll check previous original PR to confirm.

@nalind PTAL

**Comment by rhatdan** [comment]
@nalind @giuseppe @vrothberg PTAL

File: tests/bud.bats [hunk 1]
```diff
@@ -456,13 +456,28 @@ symlink(subdir)"
...
+FROM oci-${target}
+ENV date="tomorrow"
+_EOF
+  run_buildah build --format docker --unsetenv PATH --unsetenv foo --signature-policy ${TESTSDIR}/policy.json -t docker-${target} -f $mytmpdir/Containerfile .
   run_buildah inspect --type=image --format '{{.OCIv1.Config.Env}}' docker-${target}
-  expect_output "[]" "No Path should be defined"
+  expect_output "[container=buildah date=tomorrow]" "No Path should be defined"
+  run_buildah inspect --type=image --format '{{.Docker.Config.Env}}' docker-${target}
+  expect_output "[container=buildah date=tomorrow]" "No Path should be defined"
+
```
Comment (nalind) [inline tests/bud.bats:480]: Remove extra blank line.

**Comment by nalind** [comment]
Can we add a check that ensures two builds with `--layers`, one with a `--unsetenv` and one without, produce different images, i.e., that one doesn't get mistakenly used as the cached result of the other?

**Pull request merged**